stdlib/test/builtin/test_sort.mojo

```diff
@@ -10,589 +10,245 @@
+    return result
+
+
+fn assert_sorted[D: DType](inout list: List[SIMD[D, 1]]) raises:
+    sort[D](list)
+    for i in range(1, len(list)):
+        assert_true(
+            list[i] >= list[i - 1], str(list[i - 1]) + " > " + str(list[i])
+        )
+
+
+fn assert_sorted_string(inout list: List[String]) raises:
+    for i in range(1, len(list)):
+        assert_true(
+            list[i] >= list[i - 1], str(list[i - 1]) + " > " + str(list[i])
+        )
+
+
+fn assert_sorted[D: ComparableCollectionElement](inout list: List[D]) raises:
+    for i in range(1, len(list)):
+        assert_true(list[i] >= list[i - 1], "error at index: " + str(i))
 
-# RUN: %mojo %s | FileCheck %s
-from collections import List
-from random import random_si64, seed
-
-from builtin.sort import _quicksort, _small_sort
-
-
-# CHECK-LABEL: test_sort_small_3
-fn test_sort_small_3():
-    print("== test_sort_small_3")
-    alias length = 3
-
-    var list = List[Int]()
-
-    list.append(9)
-    list.append(1)
-    list.append(2)
-
-    @parameter
-    fn _less_than_equal[type: AnyRegType](lhs: type, rhs: type) -> Bool:
-        return rebind[Int](lhs) <= rebind[Int](rhs)
-
-    var ptr = rebind[Pointer[Int]](list.data)
-    _small_sort[length, Int, _less_than_equal](ptr)
-
-    # CHECK: 1
-    # CHECK: 2
-    # CHECK: 9
-    for i in range(length):
-        print(list[i])
-
-
-# CHECK-LABEL: test_sort_small_5
-fn test_sort_small_5():
-    print("== test_sort_small_5")
-    alias length = 5
-
-    var list = List[Int]()
-
-    list.append(9)
-    list.append(1)
-    list.append(2)
-    list.append(3)
-    list.append(4)
-
-    @parameter
-    fn _less_than_equal[type: AnyRegType](lhs: type, rhs: type) -> Bool:
-        return rebind[Int](lhs) <= rebind[Int](rhs)
-
-    var ptr = rebind[Pointer[Int]](list.data)
-    _small_sort[length, Int, _less_than_equal](ptr)
-
-    # CHECK: 1
-    # CHECK: 2
-    # CHECK: 3
-    # CHECK: 4
-    # CHECK: 9
-    for i in range(length):
-        print(list[i])
-
-
-# CHECK-LABEL: test_sort0
-fn test_sort0():
-    print("== test_sort0")
-
-    var list = List[Int]()
-
-    sort(list)
-
-
-# CHECK-LABEL: test_sort2
-fn test_sort2():
-    print("== test_sort2")
-
-    alias length = 2
-    var list = List[Int]()
-
-    list.append(-1)
-    list.append(0)
-
-    sort(list)
-
-    # CHECK: -1
-    # CHECK: 0
-    for i in range(length):
-        print(list[i])
-
-    list[0] = 2
-    list[1] = -2
-
-    sort(list)
-
-    # CHECK: -2
-    # CHECK: 2
-    for i in range(length):
-        print(list[i])
-
-
-# CHECK-LABEL: test_sort3
-fn test_sort3():
-    print("== test_sort3")
-
-    alias length = 3
-    var list = List[Int]()
-
-    list.append(-1)
-    list.append(0)
-    list.append(1)
-
-    sort(list)
-
-    # CHECK: -1
-    # CHECK: 0
-    # CHECK: 1
-    for i in range(length):
-        print(list[i])
 
-    list[0] = 2
-    list[1] = -2
-    list[2] = 0
+def test_sort_random_numbers_u8():
```

Comment:
**Suggestion** Since the next 100 lines are repeated with the whole parametric thing being the DType, can we use a type list and instantiate a common test function parameterized over said `DType` instead? See https://github.com/modularml/mojo/blob/f5f5109541c31615a68a3c4b58bd1e75b59625f6/stdlib/src/builtin/dtype.mojo#L739 which has an example of putting all the dtypes in a type list and running a function over it to iterate over each type in the type list.  It would simplify this test code a fair bit IMO.